### ISSUE TITLE
Firmware delete

### ISSUE THREAD

**Issue by torabi12** [e-m-b-a/embark#38]
**Describe the bug**
If I delete an uploaded firmware the EMBArk web page displays error messages.

**To Reproduce**
1. EMBArk installation (default mode)
2. Start EMBArk: sudo ./run-server.sh
3. Upload any FW file
4. Do a scan
5. Try to delete the uplaoded FW
6. See error

**Expected behavior**
Deletion should be working.

**Screenshots**
![kép](https://user-images.githubusercontent.com/98665813/162818753-7174bd22-ada6-4f97-b08d-24a460df19c5.png)

**Desktop (please complete the following information):**
 - OS: [Kali Linux 2022.01]

**Comment by m-1-k-3** [comment]
Thank you for reporting. As soon as you are using the firmware for an analysis the firmware is used and locked. But you are right, we should include a nice message for this.

**Comment by torabi12** [comment]
The scan has been finished and nothing was running. I just went back the the upload page and selected the 1st file and wanted to delete it and got the above error. After I pushed the back button in the browser EMBArk page came back with a popup saying the firmware is deleted but the file remained on the list and the delete was not done. It is easy to reproduce but this was my experience.

**Comment by m-1-k-3** [comment]
As soon as you have a scan of the firmware this firmware file is used and can't be deleted anymore.

**Comment by torabi12** [comment]
I thought the delete function can be used for any uploaded file, after 20 or more scans the list is very long and it would be great the maintain the environment and drop out the obsolete bins.
I think you should somehow display a message for this function or consider just to remove it. This is only my opinion, maybe you will have a better solution.

Thanks and regards,
Torabi

**Comment by m-1-k-3** [comment]
for sure we need to improve the firmware handling process in the future. There are also multiple other features around this topic missing. I will mark this issue as enhancement.

**Comment by torabi12** [comment]
Hello Benedikt,

I checked again the firmware delete feature but it is not working on my side. I did an EMBArk scan and I tried to delete it but I got an error, I logged out, stopped the server., restarted it, logged in tried again to delete but same error came:
![kép](https://user-images.githubusercontent.com/98665813/168418120-2fbb632e-a2ca-466f-a9e0-b692bd3bba95.png)

It is a brand new installation Kali 2022.1 with apt full-upgrade before the EMBArk install and the installation was without any error.

Does the FW delete work on your side? Or may I did something wrong?

Regards,
Torabi
